relay-server/src/services/store.rs

```diff
@@ -914,11 +914,6 @@ impl StoreService {
             }
         };
 
-        if let Some(measurements) = &mut span.measurements {
-            measurements
-                .retain(|_, v| v.as_ref().and_then(|v| v.value).is_some_and(f64::is_finite));
-        }
```

Comment:
#4828 changed measurements to `FiniteF64`, which means this validation is unnecessary.